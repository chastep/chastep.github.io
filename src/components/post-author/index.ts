export { PostAuthor } from "./post-author";
