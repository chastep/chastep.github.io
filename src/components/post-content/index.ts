export { PostContent } from "./post-content";
