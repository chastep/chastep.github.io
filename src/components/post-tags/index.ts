export { PostTags } from "./post-tags";