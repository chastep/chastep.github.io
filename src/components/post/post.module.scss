@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.post {
  grid-column: span 12;

  .content {
    @include mixins.margin-auto();
  }

  .footer {
    max-width: variables.$layout-post-width;
    @include mixins.padding(0, 0.5);
    @include mixins.margin-auto();
  }

  .comments {
    max-width: variables.$layout-post-width;
    @include mixins.padding(0, 0.5);
    @include mixins.margin-auto();
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: center;
    text-align: center;
    @include mixins.margin(1, 0, 0, 0);

    .buttonArticles {
      @include mixins.margin(0, 0.5, 0, 0);
    }
  }
}

@include mixins.breakpoint-lg {
  .post {
    .footer {
      @include mixins.padding-equal(0);
    }

    .comments {
      @include mixins.padding-equal(0);
    }

    .buttons {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include mixins.margin(0, 0.5, 0, 0);
    }
  }
}
