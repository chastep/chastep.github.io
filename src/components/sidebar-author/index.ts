export { SidebarAuthor } from "./sidebar-author";