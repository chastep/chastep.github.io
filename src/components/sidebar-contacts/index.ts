export { SidebarContacts } from "./sidebar-contacts";
