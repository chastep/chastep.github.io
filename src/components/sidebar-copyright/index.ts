export { SidebarCopyright } from "./sidebar-copyright";
