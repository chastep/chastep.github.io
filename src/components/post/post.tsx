import React, { type FC } from "react";

import type { Node } from "@/types/node";
import { Button } from "@/components/button";
import { PostTags } from "@/components/post-tags";
import { PostAuthor } from "@/components/post-author";
import { PostFooter } from "@/components/post-footer";
import { PostContent } from "@/components/post-content";
import { ThemeSwitcher } from "@/components/theme-switcher";

import * as styles from "./post.module.scss";

interface PostProps {
  post: Node;
}

const Post: FC<PostProps> = ({ post }) => {
  const { html } = post;
  const { tagSlugs } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <div className={styles.buttons}>
        <Button className={styles.buttonArticles} title="All Articles" to="/" />
        <ThemeSwitcher />
      </div>
      <div className={styles.content}>
        <PostContent body={html} title={title} />
      </div>
      <div className={styles.footer}>
        <PostFooter date={date} />
        {tags && tagSlugs && <PostTags tags={tags} tagSlugs={tagSlugs} />}
        <PostAuthor />
      </div>
    </div>
  );
};

export { Post };
