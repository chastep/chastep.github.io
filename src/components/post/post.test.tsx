import React from "react";
import { StaticQuery, useStaticQuery } from "gatsby";
import { test, describe, expect, beforeEach, mock } from "bun:test";

import * as mocks from "@/mocks";
import { Post } from "@/components/post";
import { createSnapshotsRenderer, renderWithCoilProvider } from "@/utils/render-with-coil-provider";

const mockedUseStaticQuery = useStaticQuery as ReturnType<typeof mock>;
const mockedStaticQuery = StaticQuery as unknown as ReturnType<typeof mock>;

describe("Post", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) => render(mocks.siteMetadata));
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const props = { post: mocks.markdownRemark };
    const tree = createSnapshotsRenderer(<Post {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("buttons is rendered correctly and exists", () => {
    const props = { post: mocks.markdownRemark };
    const el = renderWithCoilProvider(<Post {...props} />);
    expect(el.getByText("All Articles")).toBeInTheDocument();
  });
});
