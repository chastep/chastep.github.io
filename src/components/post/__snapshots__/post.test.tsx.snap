// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "All Articles",
          ],
          "props": {
            "aria-current": "page",
            "className": "",
            "href": "/",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "style": {},
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Perfecting the Art of Perfection",
              ],
              "props": {
                "className": undefined,
              },
              "type": "h1",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": undefined,
                "dangerouslySetInnerHTML": {
                  "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
                },
              },
              "type": "div",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Published",
                " ",
                "Sep 1, 2016",
              ],
              "props": {
                "className": undefined,
              },
              "type": "p",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": [
                        "John Doe",
                      ],
                      "props": {},
                      "type": "strong",
                    },
                    " on X",
                  ],
                  "props": {
                    "className": undefined,
                    "href": "https://www.x.com/#",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "type": "a",
                },
              ],
              "props": {
                "className": undefined,
              },
              "type": "p",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
